pkg/client/clientset_v1alpha1/typed/nodeset/v1alpha1/nodeset.go

```diff
@@ -0,0 +1,172 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	types "k8s.io/apimachinery/pkg/types"
+	watch "k8s.io/apimachinery/pkg/watch"
+	rest "k8s.io/client-go/rest"
+	scheme "kube-node/nodeset/pkg/client/clientset_v1alpha1/scheme"
+	v1alpha1 "kube-node/nodeset/pkg/nodeset/v1alpha1"
+)
+
+// NodeSetsGetter has a method to return a NodeSetInterface.
+// A group's client should implement this interface.
+type NodeSetsGetter interface {
+	NodeSets(namespace string) NodeSetInterface
```

Comment:
The new TPRs (called CustomResources) – being alpha in 1.7 – will support cluster wide resources. We will have to switch to them at some point. We can ignore the namespace here for now.